### PR TITLE
Fix figma library alert 

### PR DESCRIPTION
**Pull request by dgtlntv**
## Done
Fixed the Figma library alert which [is failing](https://github.com/canonical/vanilla-framework/actions/runs/9552881621/job/26330491088). It seems like the issue was with the manually added newlines and json encoding.

## QA

- But the release notes you want to test with in a release.json like so: 
```
{
  "release": {
    "body": "## Getting Vanilla Framework\r\n\r\nInstall with NPM: https://www.npmjs.com/package/vanilla-framework\r\nVisit the documentation at https://vanillaframework.io/docs\r\n\r\n## New in Vanilla v4.13.0\r\n\r\n## \uD83D\uDE80 Features\r\n\r\n- Add new image container component and deprecate legacy image variants by @jmuzina (#5097)\r\n\r\n## \uD83D\uDC1B Bug Fixes\r\n\r\n- Fix ascending/descending icons on sortable tables by @pastelcyborg (#5135)\r\n- Fix application layout tooltip overflow by @pastelcyborg (#5134)\r\n\r\n## \uD83D\uDCDD Documentation\r\n\r\n- Update Pull Requests README with additional information and steps by @pastelcyborg (#5137)\r\n\r\n## \uD83D\uDD28 Maintenance\r\n\r\n- Update accordion pattern to use new color theme variables by @jmuzina (#5138)\r\n- Update Matrix component with new theme color variables by @pastelcyborg (#5136)\r\n- Notify Figma library maintainers on release by @dgtlntv (#5132)\r\n- Update dependency @canonical/cookie-policy to v3.6.4 by @renovate (#5131)\r\n- Update article pagination to use new theme variables by @jmuzina (#5126)\r\n- Update lists pattern to use new color theme variables by @jmuzina (#5124)\r\n- Update Pull Quote component with new theme color variables by @pastelcyborg (#5133)\r\n- Update alternative background colour in dark theme by @bartaz (#5128)\r\n- Update card pattern to support color themes by @jmuzina (#5112)\r\n- Bump braces from 3.0.2 to 3.0.3 by @dependabot (#5121)\r\n- Update Table of Contents to use new color theme variables by @pastelcyborg (#5113)\r\n- Reduce padding on navigation items by @bartaz (#5116)\r\n\r\n**Full Changelog**: https://github.com/canonical/vanilla-framework/compare/v4.12.0...v4.13.0"
  }
}
```
- Run the following act command: `act --job notify-figma -e release.json --container-architecture linux/amd64 -s FIGMA_LIBRARY_ALERT_URL="the-webbot-url"`
- See if that runs without errors and posts a message to the Figma library maintainers chat

### CODE REVIEW

File: .github/workflows/publish-on-release.yml [hunk 1]
```diff
@@ -74,11 +74,21 @@ jobs:
...
-          echo "change_summary=$CHANGE_SUMMARY" >> $GITHUB_OUTPUT
+          CHANGE_SUMMARY=$(cat <<EOF
+          
+          $(echo "$RELEASE_NOTES" | sed -n '/## 🚀 Features/,/## /p' | sed '$d')
+
+          $(echo "$RELEASE_NOTES" | sed -n '/## 🐛 Bug Fixes/,/## /p' | sed '$d')
+
+          $(echo "$RELEASE_NOTES" | sed -n '/## 🔨 Maintenance/,/## /p' | sed '$d')
+          EOF
+          )
```
Comment (jmuzina) [inline .github/workflows/publish-on-release.yml:85]: Something is still needed to escape/sanitize these: I was able to execute arbitrary commands on the runner with this input:

```json
{
  "release": {
    "body": "## Getting Vanilla Framework\r\n\r\nInstall with NPM: https://www.npmjs.com/package/vanilla-framework\r\nVisit the documentation at https://vanillaframework.io/docs\r\n\r\n## New in Vanilla v4.13.0\r\n\r\n## \uD83D\uDE80 Features\r\n\r\n- Add new image container component and deprecate legacy image variants by @jmuzina (#5097)\r\n\r\n## \uD83D\uDC1B Bug Fixes\r\n\r\n- Fix ascending/descending icons on sortable tables by @pastelcyborg (#5135)\r\n- Fix application layout tooltip overflow by @pastelcyborg (#5134)\r\n\r\n## \uD83D\uDCDD Documentation\r\n\r\n- Update Pull Requests README with additional information and steps by @pastelcyborg (#5137)\r\n\r\n## \uD83D\uDD28 Maintenance\r\n\r\n- Update accordion pattern to use new color theme variables by @jmuzina (#5138)\r\nThis is a test of a bash command: echo $(whoami)- Update Matrix component with new theme color variables by @pastelcyborg (#5136)\r\n- Notify Figma library maintainers on release by @dgtlntv (#5132)\r\n- Update dependency @canonical/cookie-policy to v3.6.4 by @renovate (#5131)\r\n- Update article pagination to use new theme variables by @jmuzina (#5126)\r\n- Update lists pattern to use new color theme variables by @jmuzina (#5124)\r\n- Update Pull Quote component with new theme color variables by @pastelcyborg (#5133)\r\n- Update alternative background colour in dark theme by @bartaz (#5128)\r\n- Update card pattern to support color themes by @jmuzina (#5112)\r\n- Bump braces from 3.0.2 to 3.0.3 by @dependabot (#5121)\r\n- Update Table of Contents to use new color theme variables by @pastelcyborg (#5113)\r\n- Reduce padding on navigation items by @bartaz (#5116)\r\n\r\n**Full Changelog**: https://github.com/canonical/vanilla-framework/compare/v4.12.0...v4.13.0"
  }
}
```

There is an `This is a test of a bash command: echo $(whoami)` embedded in the Maintenance section here. This causes an output that contains the result of the bash execution:
```
This is a test of a bash command: echo root
```

I tried to use this payload to exfiltrate the figma library url secret:
```json
{
  "release": {
    "body": "## Getting Vanilla Framework\r\n\r\nInstall with NPM: https://www.npmjs.com/package/vanilla-framework\r\nVisit the documentation at https://vanillaframework.io/docs\r\n\r\n## New in Vanilla v4.13.0\r\n\r\n## \uD83D\uDE80 Features\r\n\r\n- Add new image container component and deprecate legacy image variants by @jmuzina (#5097)\r\n\r\n## \uD83D\uDC1B Bug Fixes\r\n\r\n- Fix ascending/descending icons on sortable tables by @pastelcyborg (#5135)\r\n- Fix application layout tooltip overflow by @pastelcyborg (#5134)\r\n\r\n## \uD83D\uDCDD Documentation\r\n\r\n- Update Pull Requests README with additional information and steps by @pastelcyborg (#5137)\r\n\r\n## \uD83D\uDD28 Maintenance\r\n\r\n- Update accordion pattern to use new color theme variables by @jmuzina (#5138)\r\nThis is a test of a bash command: echo $(whoami); env vars are echo $(printenv); figma library url is ${{ secrets.FIGMA_LIBRARY_ALERT_URL }} - Update Matrix component with new theme color variables by @pastelcyborg (#5136)\r\n- Notify Figma library maintainers on release by @dgtlntv (#5132)\r\n- Update dependency @canonical/cookie-policy to v3.6.4 by @renovate (#5131)\r\n- Update article pagination to use new theme variables by @jmuzina (#5126)\r\n- Update lists pattern to use new color theme variables by @jmuzina (#5124)\r\n- Update Pull Quote component with new theme color variables by @pastelcyborg (#5133)\r\n- Update alternative background colour in dark theme by @bartaz (#5128)\r\n- Update card pattern to support color themes by @jmuzina (#5112)\r\n- Bump braces from 3.0.2 to 3.0.3 by @dependabot (#5121)\r\n- Update Table of Contents to use new color theme variables by @pastelcyborg (#5113)\r\n- Reduce padding on navigation items by @bartaz (#5116)\r\n\r\n**Full Changelog**: https://github.com/canonical/vanilla-framework/compare/v4.12.0...v4.13.0"
  }
}
```
This does not expose the secret (maybe there is some other way to expose it though) - but it does cause the extraction step to fail:
```
| **Full Changelog**: https://github.com/canonical/vanilla-framework/compare/v4.12.0...v4.13.0: bad substitution
[Publish Vanilla release/Notify Figma library maintainers]   ❌  Failure - Main Extract release notes sections
[Publish Vanilla release/Notify Figma library maintainers] exitcode '1': failure
[Publish Vanilla release/Notify Figma library maintainers] 🏁  Job failed
Error: Job 'Notify Figma library maintainers' failed
```
Does the script need to extract the full release notes? Maybe we are unnecessarily exposing ourselves to risk here - why not send a link to the release to the MM bot?

**Review by jmuzina — CHANGES_REQUESTED** [review]

File: .github/workflows/publish-on-release.yml (hunk 1 above)
Comment (jmuzina) [reply .github/workflows/publish-on-release.yml:85]: See Github Actions webhook docs: https://docs.github.com/en/rest/releases/releases?apiVersion=2022-11-28#get-a-release

The web URL to the release should be available either by [github context](https://docs.github.com/en/actions/learn-github-actions/contexts#github-context) (hopefully `${{ github.event.release.html_url }}`), or, if not like that, by using their REST API as shown [here](https://docs.github.com/en/rest/releases/releases?apiVersion=2022-11-28#get-a-release).

**Comment by dgtlntv** [comment]
To test the latest change change the release.json to
```
{
  "release": {
    "html_url": "https://github.com/canonical/vanilla-framework/releases/tag/v4.13.0"
  }
}
```

File: .github/workflows/publish-on-release.yml [hunk 2]
```diff
@@ -70,15 +70,6 @@ jobs:
...
-          RELEASE_NOTES=$(echo "${{ github.event.release.body }}" | jq -R . | jq -r .)
-          FEATURES=$(echo "$RELEASE_NOTES" | sed -n '/## 🚀 Features/,/## /p' | sed '$d')
-          BUG_FIXES=$(echo "$RELEASE_NOTES" | sed -n '/## 🐛 Bug Fixes/,/## /p' | sed '$d')
-          MAINTENANCE=$(echo "$RELEASE_NOTES" | sed -n '/## 🔨 Maintenance/,/## /p' | sed '$d')
-          CHANGE_SUMMARY="\n$FEATURES\n\n$BUG_FIXES\n\n$MAINTENANCE\n"
-          echo "change_summary=$CHANGE_SUMMARY" >> $GITHUB_OUTPUT
       - name: Notify Figma library maintainers
         run: |
-          curl -X POST -H "Content-Type: application/json" -d '{"source": "Vanilla CSS", "change-summary": "${{ steps.extract_release_notes.outputs.change_summary }}"}' ${{ secrets.FIGMA_LIBRARY_ALERT_URL }}
+          curl -X POST -H "Content-Type: application/json" -d '{ "source": "Vanilla CSS", "change-summary": " [Vanilla release notes](${{ github.event.release.html_url }})" }' ${{ secrets.FIGMA_LIBRARY_ALERT_URL }}
```
Comment (bartaz) [inline .github/workflows/publish-on-release.yml:75]: Maybe later we can make it nicer, at least by adding version number or something. But let's test the simple one first.

**Review by bartaz — APPROVED** [review]
Looks good. Let's test it out with next release.

**Pull request merged**